src/com/trollworks/gcs/spell/SpellTechnique.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 1998-2020 by Richard A. Wilkes. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public License, version 2.0.
+ * If a copy of the MPL was not distributed with this file, You can obtain one at
+ * http://mozilla.org/MPL/2.0/.
+ *
+ * This Source Code Form is "Incompatible With Secondary Licenses", as defined by the
+ * Mozilla Public License, version 2.0.
+ */
+
+package com.trollworks.gcs.spell;
+
+import com.trollworks.gcs.character.GURPSCharacter;
+import com.trollworks.gcs.common.DataFile;
+import com.trollworks.gcs.common.LoadState;
+import com.trollworks.gcs.skill.Skill;
+import com.trollworks.gcs.skill.SkillDefault;
+import com.trollworks.gcs.skill.SkillDefaultType;
+import com.trollworks.gcs.skill.SkillDifficulty;
+import com.trollworks.gcs.skill.SkillLevel;
+import com.trollworks.gcs.skill.Technique;
+import com.trollworks.gcs.template.Template;
+import com.trollworks.gcs.widgets.outline.ListRow;
+import com.trollworks.gcs.widgets.outline.RowEditor;
+import com.trollworks.toolkit.io.xml.XMLReader;
+import com.trollworks.toolkit.utility.I18n;
+
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.HashSet;
+import java.util.Set;
+
+/** A GURPS Spell for the Ritual Magic system. */
+public class SpellTechnique extends Spell {
```

Comment:
Since these seem to be named "Ritual Magic Spell" or something similar throughout the interface, the class should and xml tags should probably reflect that as well

Reply:
Would SpellRitualMagic, SpellRitualMagicEditor be ok?

Reply:
I'd prefer `RitualMagicSpell`, `RitualMagicSpellEditor`, etc. so that it reads naturally.